drivers/can/can_stm32h7.c

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) 2022 Blue Clover
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/can.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <drivers/clock_control.h>
+#include <drivers/pinctrl.h>
+#include <kernel.h>
+#include <stm32_ll_rcc.h>
+#include "can_mcan.h"
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(can_driver, CONFIG_CAN_LOG_LEVEL);
+
+#define DT_DRV_COMPAT st_stm32h7_fdcan
+
+struct can_stm32h7_config {
+	struct can_mcan_msg_sram *msg_sram;
+	void (*config_irq)(void);
+	struct can_mcan_config mcan_cfg;
+	const struct pinctrl_dev_config *pcfg;
+	struct stm32_pclken pclken;
+};
+
+struct can_stm32h7_data {
+	struct can_mcan_data mcan_data;
+};
+
+int can_stm32h7_get_core_clock(const struct device *dev, uint32_t *rate)
```

Comment:
All functions only used in this file must be `static`. I thought I commented on this earlier on, but I cannot find that comments now. Otherwise looks fine.

Reply:
Now that you mention it, they definitely should be! Fixed.